services/autoscaling/src/simcore_service_autoscaling/modules/dask.py

```diff
@@ -118,19 +81,87 @@ def _find_by_worker_host(
         dask_worker: tuple[DaskWorkerUrl, DaskWorkerDetails]
     ) -> bool:
         _, details = dask_worker
-        return bool(details["host"] == node_ip) or bool(
-            node_hostname in details["name"]
-        )
+        if match := re.match(DASK_NAME_PATTERN, details["name"]):
+            return bool(match.group("private_ip") == node_hostname)
+        return False
 
     filtered_workers = dict(filter(_find_by_worker_host, workers.items()))
     if not filtered_workers:
         raise DaskWorkerNotFoundError(
             worker_host=ec2_instance.aws_private_dns, url=client.scheduler.address
         )
-    assert len(filtered_workers) == 1  # nosec
+    assert (
+        len(filtered_workers) == 1
+    ), f"returned workers {filtered_workers}, {node_hostname=}"  # nosec
     return next(iter(filtered_workers.items()))
 
 
+async def is_worker_connected(
+    scheduler_url: AnyUrl, worker_ec2_instance: EC2InstanceData
+) -> bool:
+    with contextlib.suppress(DaskNoWorkersError, DaskWorkerNotFoundError):
+        async with _scheduler_client(scheduler_url) as client:
+            _dask_worker_from_ec2_instance(client, worker_ec2_instance)
+            return True
+    return False
+
+
+async def list_unrunnable_tasks(url: AnyUrl) -> list[DaskTask]:
```

Comment:
Q: unrunnable tasks are tasks that are pending because the required resources are not available?

Reply:
well unrunnable tasks is a concept of tasks that cannot run.
Now if we talk about dynamic services, these are the services that are pending (in docker world)
in computational services, these are dask tasks that have currently _no workers_ which means no worker is able to take them (properly unrunnable) AND tasks that a worker took in (in a way queued right?) but the problem is that the dask-scheduler does not know if a worker queued them or is actually processing them...